src/equisolve/utils/split_data.py

```diff
@@ -0,0 +1,431 @@
+        Used to control shuffling of the unique indices, which dictate the data
+        that ends up in each of the split output tensors. If None, no shuffling
+        of the indices occurs. If -1, shuffling occurs but with no random seed
+        set. If an integer != -1, shuffling is executed but with a random seed
+        set to this value.
+
+    :return split_tensors: list of list of :py:class:`TensorMap`. The ith
+        element in the list contains `n_groups` :py:class:`TensorMap` objects
+        corresponding to the split ith :py:class:`TensorMap` of the input list
+        `tensors`.
+    :return grouped_labels: list of :py:class:`Labels` corresponding to the
+        unique indices according to the specified `axis` and `names` that are
+        present in each of the returned groups of :py:class:`TensorMap`. The
+        length of this list is `n_groups`.
+
+    Examples
+    --------
+
+    Split a TensorMap `tensor` into 2 new TensorMaps along the "samples" axis
+    for the "structure" metadata. Without specifying `group_sizes`, the data
+    will be split equally by structure index. If the number of unique strutcure
+    indices present in the input data is not exactly divisible by `n_groups`,
+    the group sizes will be made to the nearest int. Without specifying
+    `random_seed`, no shuffling of the structure indices will occur and they
+    will be grouped in lexigraphical order. For instance, if the input tensor
+    has structure indices 0 -> 9 (inclusive), the first new tensor will contain
+    only structure indices 0 -> 4 (inc.) and the second will contain only 5 -> 9
+    (inc).
+
+    .. code-block:: python
+
+        from equisolve.utils import split_data
+
+        [[new_tensor_1, new_tensor_2]], grouped_labels = split_data(
+            tensors=tensor,
+            axis="samples",
+            names=["structure"],
+            n_groups=2,
+        )
+
+    Split 2 tensors corresponding to input and output data into train and test
+    data, with a relative 80:20 ratio. If both input and output tensors contain
+    structure indices 0 -> 9 (inclusive), the `in_train` and `out_train` tensors
+    will contain structure indices 0 -> 7 (inc.) and the `in_test` and
+    `out_test` tensors will contain structure indices 8 -> 9 (inc.). As we want
+    to specify relative group sizes, we will pass `group_sizes` as a list of
+    float. Specifying the `random_seed` will shuffle the structure indices
+    before the groups are made.
+
+    .. code-block:: python
+
+        from equisolve.utils import split_data
+
+        [[in_train, in_test], [out_train, out_test]], grouped_labels = split_data(
+            tensors=[input, output],
+            axis="samples",
+            names=["structure"],
+            n_groups=2,                  # for train-test split
+            group_sizes=[0.8, 0.2],  # relative, a 80% 20% train-test split
+            random_seed=100,
```

Comment:
Isn't this an overkill if we also can give a list of tensors. It might be simpler for our code and it is not a big burden if the user just has to call the method twice.

Reply:
(Commented on below)

Reply:
Discussed in person.